### PR TITLE
[XPU] Add transfilter when conv2d op dilation > 1 

### PR DESCRIPTION
**Pull request by csy0225**
### PR types

Performance optimization
### PR changes

Others
### Description

When conv2d op's dilation > 1, conv2d_fusion kernel will call xdnn kernel of trans_filter, it costs too much time. So we do transfilter phase in pass to reduce the execute time.
eg: 
when ksize=[3,3], dilations = [2,2] , after the pass apply , ksize = [5, 5], dilations = [1,1]

### CODE REVIEW

File: paddle/fluid/framework/ir/do_trans_filter_pass.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+namespace framework {
+class Scope;
+}  // namespace framework
+}  // namespace paddle
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+class DoTransFilterPass : public FusePassBase {
```
Comment (hong19860320) [inline paddle/fluid/framework/ir/do_trans_filter_pass.h]: 名称改一下吧Conv2dDilation2x2To1x1TransformPass

File: paddle/fluid/inference/analysis/passes/convert_to_mixed_precision.cc [hunk 2]
```diff
@@ -89,6 +90,12 @@ void ConvertToMixedPrecisionPass::Run() {
 
   framework::ir::ConstantFoldingPass constant_folding_pass;
   constant_folding_pass.Apply(main_graph_.get());
+
+  if (backend_ == phi::Backend::XPU) {
```
Comment (hong19860320) [inline paddle/fluid/inference/analysis/passes/convert_to_mixed_precision.cc]: 为什么一定需要在 convert_to_mixed_precision 内？为啥不能放在 xpu passbuilder？

File: paddle/fluid/framework/ir/do_trans_filter_pass.h (hunk 1 above)
Comment (csy0225) [reply paddle/fluid/framework/ir/do_trans_filter_pass.h]: 已修改

File: paddle/fluid/inference/analysis/passes/convert_to_mixed_precision.cc (hunk 2 above)
Comment (csy0225) [reply paddle/fluid/inference/analysis/passes/convert_to_mixed_precision.cc]: 恩，这个fp16类型已经放在 pass 里面支持了

**Review by hong19860320 — APPROVED** [review]
LGTM

**Review by zyfncg — APPROVED** [review]

**Pull request merged**